src/connection/Form.ts

```diff
@@ -91,14 +91,34 @@ export default class Form {
 
       this.items.forEach((fields, itemIndex) => {
          if (fields.length !== this.reportedFields.length) {
-            throw new InvalidParameterError(`Item ${itemIndex} does not contain all "reported" fields.`);
+             try{
+                 this.reportedFields.forEach((field, index) => {
+                    let i=0;
+                    let found=false;
+                    for (;i<fields.length;i++)
+                    {
+                        if (this.reportedFields[index].getName() === fields[i].getName()) {
+                            found=true;
+                            break;
+                        }
+                    }
+
+                    if (!found) {
+                       throw new InvalidParameterError(`Item ${itemIndex} does not contain all "reported" fields.`);
```

Comment:
Why do we need the catch block, if we throw an error at this point?

Reply:
ok it was to late :D